bundles/org.eclipse.ui.workbench/eclipseui/org/eclipse/ui/internal/Workbench.java

```diff
@@ -2714,12 +2721,88 @@ public static Object setRestartArguments(String workspacePath) {
 		if (command_line == null) {
 			return IApplication.EXIT_OK;
 		}
+		Path selectedWorkspace = Path.of(workspacePath);
+		try {
+			String workspaceLock = getWorkspaceLockDetails(selectedWorkspace.toUri().toURL());
+			if (workspaceLock != null && !workspaceLock.isEmpty()) {
+				String lockMessage = NLS.bind(WorkbenchSWTMessages.IDEApplication_workspaceCannotLockMessage2,
+						workspacePath);
+				String wsLockedError = lockMessage + System.lineSeparator() + System.lineSeparator()
+						+ NLS.bind(WorkbenchSWTMessages.IDEApplication_workspaceLockMessage,
+						workspaceLock);
+
+				MessageDialog.openError(PlatformUI.getWorkbench().getActiveWorkbenchWindow().getShell(),
```

Comment:
The code is also used from `IDEApplication.checkInstanceLocation()` where no active workbench window exists.